### PR TITLE
[7050] Return Study Mode in API Responses

### PR DESCRIPTION
**Pull request by AdColvin**
### Context
The study mode for the course wasn't being returned in API responses, even when it was provided in the request.

### Changes proposed in this pull request
It turns out that the study mode wasn't being saved correctly when the trainee was created. So this adds the missing method to the mapping class so that it now gets included. 

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

File: spec/requests/api/v0.1/post_hesa_trainees_spec.rb [hunk 1]
```diff
@@ -77,6 +77,10 @@
       expect(Trainee.last.state).to eq("submitted_for_trn")
     end
 
+    it "sets the correct study_mode" do
```
Comment (Reyko) [inline spec/requests/api/v0.1/post_hesa_trainees_spec.rb:80]: Nice! Could you add this to `put_trainee_spec.rb` as well?

**Comment by AdColvin** [comment]
@Reyko @stevehook thanks for the reviews. Actually on checking the update logic, we were overriding the hesa_trainee_detail attributes on every update. So if we didn't provide any values for the attributes there, they would all be set to nil. So I fixed that in [this commit](https://github.com/DFE-Digital/register-trainee-teachers/pull/4251/commits/acd24e09a782c2b639afd274d19a7b81bbe5ab57) and added a test, let me know what you think 🙏

File: spec/requests/api/v0.1/put_trainee_spec.rb [hunk 2]
```diff
@@ -199,6 +199,7 @@
 
             expect(response).to have_http_status(:ok)
             expect(trainee.first_names).to eq("Alice")
+            expect(response.parsed_body[:data][:study_mode]).to eq("63")
```
Comment (tomtrentham) [inline spec/requests/api/v0.1/put_trainee_spec.rb:202]: I can't see where this value is being updated in the PUT request.

File: app/models/api/trainee_attributes/v01.rb [hunk 3]
```diff
@@ -126,17 +126,22 @@ def assign_attributes(attributes)
...
           trainee_disabilities_attributes << { disability_id: disability.id }
         end
       end
 
+      def update_hesa_trainee_detail_attributes(attributes)
+        existing_hesa_attributes = hesa_trainee_detail_attributes&.attributes || {}
+        new_hesa_attributes = HesaTraineeDetailAttributes::V01.new(
+          attributes.slice(*HesaTraineeDetailAttributes::V01::ATTRIBUTES),
+        )
+        self.hesa_trainee_detail_attributes = new_hesa_attributes.assign_attributes(existing_hesa_attributes)
```
[the remark this answers is not in the document]
Comment (AdColvin) [reply app/models/api/trainee_attributes/v01.rb]: I don't think it makes a difference, but I agree it reads better the other way around. I've changed it 👍

File: spec/requests/api/v0.1/put_trainee_spec.rb (hunk 2 above)
Comment (AdColvin) [reply spec/requests/api/v0.1/put_trainee_spec.rb:210]: Updated 👍

**Comment by defong** [comment]
> @Reyko @stevehook thanks for the reviews. Actually on checking the update logic, we were overriding the hesa_trainee_detail attributes on every update. So if we didn't provide any values for the attributes there, they would all be set to nil. So I fixed that in [this commit](https://github.com/DFE-Digital/register-trainee-teachers/pull/4251/commits/acd24e09a782c2b639afd274d19a7b81bbe5ab57) and added a test, let me know what you think 🙏

I also wrote a test but on a lower level https://github.com/DFE-Digital/register-trainee-teachers/pull/4249/commits/ca968981c83a5bbb61ad9cf985d228937ce7842a

**Review by tomtrentham — APPROVED** [review]
This is now returning the `study_mode` consitently 👍

**Review by Reyko — APPROVED** [review]

**Pull request merged**